flow/record/adapter/elastic.py

```diff
@@ -138,21 +149,24 @@ def flush(self) -> None:
         pass
 
     def close(self) -> None:
+        self.queue.put(StopIteration)
+        self.event.wait()
```

Comment:
note that `self.queue` and `self.event` here might not exist yet if our `__init__` raises an exception, which results that this `close()` method is called due to a finally block and causing an AttributeError as they are not initialised yet.

That was the reason why we added `if hasattr(self, "es")` guard, which in hindsight is not very clean.

I think it would be safe to move `self.queue = ...` and `self.event = ...` higher up (before `self.es = ...`) in the `__init__` and it would be OK.